lib/ipmeta.c

```diff
@@ -194,6 +194,53 @@ ipmeta_provider_t **ipmeta_get_all_providers(ipmeta_t *ipmeta)
   return ipmeta->providers;
 }
 
+void ipmeta_record_clear(ipmeta_record_t *record)
+{
+  if (record == NULL) {
+    return;
+  }
+  free(record->region);
```

Comment:
since this is a "clear" method, we probably need to do a memset zero on the non-dynamic members of the record at the end of this function.
and then all of the `record->XXX = NULL` statements can be removed.